test/extensions/filters/http/wasm/test_data/test_grpc_stream_cpp.cc

```diff
@@ -107,16 +107,18 @@ FilterHeadersStatus GrpcStreamContext::onRequestHeaders(uint32_t, bool) {
   HeaderStringPairs initial_metadata;
   if (root()->grpcStreamHandler("bogus service string", "service", "method", initial_metadata,
                                 std::unique_ptr<GrpcStreamHandlerBase>(
-                                    new MyGrpcStreamHandler())) != WasmResult::ParseFailure) {
-    logError("unexpected bogus service string OK");
+                                    new MyGrpcStreamHandler())) == WasmResult::ParseFailure) {
+    logError("unexpected bogus service parse failure");
```

Comment:
Nit: This is expected failure, so you should remove the "unexpected" word from the log line.